engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -58,7 +58,7 @@ def verify_product_activation
 
           if product.base?
             verify_base_product_activation(product)
-          else
+          elsif !product.free? && params[:token].blank?
```

Comment:
How does this implement de-registration?

Reply:
it is not directly related to the de-registration but it is needed for the tests

The change is implemented [here](https://github.com/SUSE/rmt/pull/1203/files#diff-7edbe8015a11c2bffc38e12ec3aa8e97c68e66201ef3bf64947d3bad213ea1e1R61) being part of #1203

once #1203 gets merged, it will get merged here and that change will be gone from the diff

Reply:
Let's merge #1203 then to avoid confusion and duplication in this pr.

Reply:
merged